sdk/resourcemanager/Azure.ResourceManager.Core/src/ArmException.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Runtime.Serialization;
+using Azure.Core;
+
+#nullable enable
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// A class representing exceptions thrown by Azure ARM services.
+    /// </summary>
+    [Serializable]
+    public class ArmException : RequestFailedException
+    {
+        /// <summary>
+        /// Gets and set the Code of the ARM exception.
+        /// </summary>
+        public string Code { get; internal set; }
+
+        /// <summary>
+        /// Gets and set the Target of the ARM exception.
+        /// </summary>
+        public string Target { get; internal set; }
+
+        /// <summary>
+        /// Gets and set the Details of the ARM exception.
+        /// </summary>
+        public ArmException[]? Details { get; internal set; }
```

Comment:
Can we represent this using the InnerException + AggregateException type?

Reply:
Sounds good.

Reply:
Implementation do-able. Though need decision on the point raised on AdditionalInfo as well.